### ISSUE TITLE
Frontpage re-design

### ISSUE THREAD

**Issue by tlienart** [JuliaLang/www.julialang.org#800]
Following #791 and a discussion on Slack, here are some actionable things we could work on to make some pages more ergonomic and lighter to load.

### Front page specific

* [ ] put the tools (IDE/Debugger tools) on a separate page,  access it via a new button  next to "Try Julia in the Browser"
* [ ] at the moment the buttons "Code examples" and "Try julia in the browser" lead to  the same REPL monster that takes forever to load, not a good sign, only one of the two should, the other one should show actual code directly (like it did before with the rosetta code)
* [ ] remove the Environment stuff and their heavy gifs and move it to the Community page 
* [ ] update the splash image (see #791) making sure that background images there are <100kb. Ideally generate SVG files with programs like Luxor etc and cycle through them
* [ ] make the youtube video link small (_see Go front page_)
* [ ] make the "recent blog post" be automatically generated from the last 3-5 blog posts

### Community page

The community page is a bit of a mess, too much information and not structured  enough, here are some points:

* [ ] add a table of content to help navigate the page
* [ ] begin the page with clear links to "where can I find a package that does X" + key Julia org. 
* [ ] have some examples with key packages, how to do X with Y (ft packages that have > 500 stars)
* [ ] remove the list of meetups and only have a link to the meetup aggregator

Thoughts & actionable points are welcome. Please bear in mind that at some point someone will have to do it so if you want us to do large changes (e.g. change framework) please provide a full working POC otherwise we will likely not do it.

**Comment by zdroid** [comment]
Why can't recent blog post be just... last blog post?

**Comment by cormullion** [comment]
Although I'm a big fan personally, I'm starting to think that having a slideshow/art-gallery isn't entirely a good idea. It can become too much a matter of personal opinion and taste, and the people who comment may be outliers (be they minimalists or lovers of the baroque), some want a Julia set, others hate background graphics... 

The most economical SVGs I can make are linear gradients - you can easily get this beauty for 727 bytes:

<img width="576" alt="Screenshot 2020-05-13 at 09 52 05" src="https://user-images.githubusercontent.com/52289/81792071-7349a900-94ff-11ea-8979-ca7233fad53e.png">

but as soon as I try to add a handful of straight lines it creeps up to 30K or so. 

<img width="576" alt="Screenshot 2020-05-13 at 09 55 10" src="https://user-images.githubusercontent.com/52289/81792435-f834c280-94ff-11ea-94dd-47c6990a02b2.png">

This might be something to do with how Cairo outputs SVG - probably not very efficiently. Funny how high quality SVGs start to get as big as low quality PNGs...

**Comment by tlienart** [comment]
> Why can't recent blog post be just... last blog post?

Sorry this is exactly what I meant, but it's not the case at the moment. You may know that stuff went from a Hugo backend to Franklin backend, I ported most of the functionality but there's a few things missing, one was the auto generation of the list of blogs (now done) and another one which I realised when writing the list here is the list of few X recent blog posts. It's an easy fix that I'll do, I just added it to the list to not forget.

**Comment by zdroid** [comment]
@tlienart 
1. Does "remove the Environment stuff and their heavy gifs and move it to the Community page" refer to the Ecosystem section, or to the Editors and IDEs? Editors definitely should be moved off the front-page.
2. What do you mean by YouTube video link?

**Comment by tlienart** [comment]
1: yes so this:

<img width="1212" alt="Screenshot 2020-05-13 at 11 57 48" src="https://user-images.githubusercontent.com/10897531/81799047-0e974a00-9511-11ea-80fe-dedbc856a791.png">

would be removed from the front page and arrive in some form on the community page instead (that's a suggestion). In any case it loads some massive gifs which don't really serve a purpose, and that should be removed.

The Editor/Tools stuff should go, I think, on a /tools/ page which could be linked to here:

---

<img width="615" alt="Screenshot 2020-05-13 at 11 59 38" src="https://user-images.githubusercontent.com/10897531/81799264-60d86b00-9511-11ea-9f29-fd9c6811b65f.png">

---

2: this (note: you may have it blocked if you use uBlock or similar)

---

<img width="1164" alt="Screenshot 2020-05-13 at 11 57 56" src="https://user-images.githubusercontent.com/10897531/81799306-7188e100-9511-11ea-8b0f-8b47fac59686.png">

---

I think it's potentially a good idea to have a youtube link here but it shouldn't be as big

**Comment by cormullion** [comment]
Can you add text as a background using HTML and CSS? - I find text tricky in SVG - either it gets converted to thousands of curves (-> big files) or it stays as text inside the SVG, then needs to be styled somehow with available fonts.

**Comment by zdroid** [comment]
1. The ecosystem section is what actually brought me into Julia. It's a really good selling point, as it lists all the key strengths of the language. It kinda says "Julia is awesome, and if you don't believe it, look at the awesome software built with it".
Maybe it can be condensed to be simpler and without GIFs?

2. YouTube video and that whole section may be removed altogether. After all, someone who finds out Julia for the first time isn't looking for JuliaCon. And someone who's looking for JuliaCon knows where to look.

**Comment by zdroid** [comment]
@cormullion You just make SVG image without text and put it as background to the .jumbotron div, like the way it is now.

**Comment by tlienart** [comment]
> * The ecosystem section is what actually brought me into Julia. It's a really good selling point, as it lists all the key strengths of the language. It kinda says "Julia is awesome, and if you don't believe it, look at the awesome software built with it".
>   Maybe it can be condensed to be simpler and without GIFs?

Right, so the key message is that there's lots of packages right? there's already that "Packages" section which is quite good? 

The tabs as they are now are all outdated anyway, what we could have instead is something that looks like it with some simple code on the left and the output on the right showcasing "nice examples" of how Package X can be used to do Y. This is what I suggested in the list except that I suggested it be in "community" but I agree it could be on the front page. 

Packages that could be shown there could be:

* Plots
* DataFrames
* Jump
* Diffeq
* Flux or Knet

There could maybe also be a list of the 10 most starred packages and what they do?

**Comment by zdroid** [comment]
@tlienart That's right. Packages and Ecosystem sections should be merged, somewhere along the lines of what you've proposed. Drop GIFs (maybe replace them with plain images) and update the lists of packages. But it seems that you'd arrive at something very similar to the current Ecosystem section, so why not just improve it and drop the packages one?

What do you think about dropping the JuliaCon section though? There's already a link to YouTube channel.

**Comment by tlienart** [comment]
Yeah I think I'm favourable to that. In any case we should do this in chunks of work so that we don't change everything at once and can get feedback on stuff. In order to not clash on things would you be interested in one of those items?

**Comment by zdroid** [comment]
I am interested. I'd go with removing JuliaCon section first (from the front-page). It maybe should be linked somewhere else.

**Comment by tlienart** [comment]
well it's already linked in the banner...

**Comment by tlienart** [comment]
also bootstrap 5 does not use jQuery, might be worth experimenting with it.

**Comment by zdroid** [comment]
@tlienart Users should be able to close the banner, though (and to have that preference stored in localStorage). It's a bit distracting to be displayed every time.

**Comment by cormullion** [comment]
@tlienart I uploaded a few SVG banner-type images [here](https://github.com/cormullion/graphics/tree/master/svg-banners). I'm not saying you should use any of these in particular, just they're the kind of image I can make for around 50kb or so. They've all been optimized so are probably as small as they can get.

Whether they're too dull or too eye-catching is another matter altogether... :) 🤣

**Comment by ViralBShah** [comment]
Let's open separate issues for redesign of other pages. I also have some graphics I use in my dudes that have logos of universities and books that I can add.

**Comment by tlienart** [comment]
@ViralBShah yes I think here I added "community" because some content would potentially be shifted over to that page but I agree that could be done in a second phase.

**Comment by ViralBShah** [comment]
Just taking a step back here. I think the point of the front page redesign should not to be minimal or lighter to load, but to get a newcomer to the project excited about Julia and find the things they need quickly. Within that experience, we certainly want a lighter / fast loading page.

**Comment by tlienart** [comment]
The main thing I wanted to do is a revamp of that ecosystem thing on the front page which would highlight top packages, maybe show some key code / plots

**Comment by cormullion** [comment]
@tlienart I added a few more versions of the Bowyer–Watson/Delaunay triangulation background image with varying opacities and with correct graph coloring (since someone commented that adjacent triangles had the same color... :) But perhaps it would be simpler to just have a plain colored background...🤣

**Comment by JeffreySarnoff** [comment]
The last few words of the final sentence can jump out of the surrounding text box, moving under the image.  The boxing should be enforced.

![fix](https://user-images.githubusercontent.com/1682118/82149080-3265d300-9824-11ea-8df5-accd4b4ca741.png)

**Comment by zdroid** [comment]
@JeffreySarnoff Is that really such an issue? It does very well if it's not just a few words left over, like here:

![Screenshot from 2020-05-17 16-25-49](https://user-images.githubusercontent.com/2725611/82151288-1af60c80-985b-11ea-9cf4-f160c0f833cb.png)

Anyway, that section will be redesigned per the points mentioned above, so a better design may be implemented.

**Comment by JeffreySarnoff** [comment]
It is possible to have the best of both, extending the box when the excess is small and overlaying below the image  otherwise. The hanging phrase looks unprofessional to me.

**Comment by cmcaine** [comment]
Anecdata: I really like the plots gifs and they were one of the reasons why I started using Julia.

You could make them `.webm`s if you need to cut bytes?

**Comment by cormullion** [comment]
not sure webm is supported on Apple mobile devices...

**Comment by cmcaine** [comment]
They support MP4, which would probably still be an improvement.

**Comment by zdroid** [comment]
@tlienart Last two items from front-page specific should be checked.

**Comment by ViralBShah** [comment]
I think a major next step is to overhaul the ecosystems. People (newcomers) often want to know what is a good set of well-tested robust packages for basic capabilities. And the ecosystems needs to point to that.

I think on the main page, this should translate into a collection of org logos, which when clicked take one to a page in `ecosystems/` about that org that list the solid packages, the experimental ones, and details about how to contact that project on discourse, slack, zulip, github, etc.

**Comment by apparluk** [comment]
A more minimalist Home page.

![julialang](https://user-images.githubusercontent.com/52227507/97273213-54339f00-1833-11eb-838f-ee29b6da2498.png)

The coloured circles are shifted to the left to fit in a smaller horizontal image.

See this [webpage](https://nurmi-labs.blogspot.com/) for a true perspective and better header font.

The 16x16 pixel .gif files are about 70 bytes each as opposed to exported
.png files at compression level 9.

The use of sprites are considered faster loading.

The Blog, JSoC, and Learn, content could be shifted to the Community page.

![blue](https://user-images.githubusercontent.com/52227507/97273428-a07edf00-1833-11eb-8601-12649eff78c5.gif) ![red](https://user-images.githubusercontent.com/52227507/97273521-c906d900-1833-11eb-9e69-f4aa1a87f74e.gif) ![green](https://user-images.githubusercontent.com/52227507/97273540-d0c67d80-1833-11eb-8b21-04c411c456a2.gif) ![purple](https://user-images.githubusercontent.com/52227507/97273560-d754f500-1833-11eb-90b2-6e61282f16c6.gif)

"Download" hover text for (blue 4063d8)
"Documentation" hover text for (red cb3c33)
"Community" hover text for (green 389826)
"Research" hover text for(purple 9558b2)

**Comment by cormullion** [comment]
Minimalist designs are cool and can be fun, but the downside is the challenges they pose for accessibility and discovery. Hover text can be difficult to get right on both desktop and mobile. The tiny images are good, but bigger downloads are just around the corner, so it may not save much bandwidth in the long run.

A change in branding would be a fun discussion! 😃 I think the use of those four coloured dots in a row is getting interestingly close to Julia Computing's branding...

**Comment by apparluk** [comment]
> Hover text can be difficult to get right on both desktop and mobile.

The four coloured circles would only appear on the Home page.

http://okmij.org/ftp/

[right aligned] Text would substitute for the 4 circles on other pages.

http://okmij.org/ftp/Computation/index.html

previous next start top

http://okmij.org/ftp/Computation/IO-monad-history.html

previous next up top

I do not propose however using a sidebar.

**Comment by cormullion** [comment]
I think those are possibly the dullest web pages I've ever seen. 😂 Probably not the sort of thing to get people excited about Julia...

**Comment by apparluk** [comment]
With a minimalist site attention could be allocated to the printed manual.

Rich Stevens' [books](http://www.kohala.com/start/) being suggested as regards structuring and design.

**Comment by apparluk** [comment]
This is preferable to candy floss.

- Tools
  - [Debugger](https://github.com/JuliaDebug/Debugger.jl)
  - [GPUs](https://juliagpu.org/)
  - [Profiler](https://docs.julialang.org/en/v1/manual/profile/)
  - [Revise](https://github.com/timholy/Revise.jl)

**Comment by apparluk** [comment]
_cf_: https://github.com/JuliaLang/www.julialang.org/issues/800#issuecomment-717072437

A 'JuliaLang/julia.github.io' repo could be created and described as a 'draft'.
